roles/deploy/tasks/main.yml

```diff
@@ -100,31 +97,39 @@
     - rake
     - skip_ansible_lint
 
-- debug: msg="{{ table_exists.stdout }}"
-  tags: rake
-- debug: msg="{{ table_exists.stderr }}"
-  tags: rake
-
 #--------------------------
 # Create a rollback release
 
 - name: create a database rollback version
-  command: bash -c "set -o pipefail && pg_dump -h '{{ db_host }}' -U '{{ db_user }}' '{{ db }}' | gzip > '{{ rollback_sql_path }}'"
-  args:
-    creates: "{{ rollback_sql_path }}"
+  postgresql_db:
+    name: "{{ db }}"
+    state: dump
+    target: /tmp/rollback.sql.gz
+  become: yes
+  become_user: postgres
   when: table_exists.stderr.find('does not exist') == -1
 
+- name: copy database dump
+  copy:
+    src: /tmp/rollback.sql.qz
+    dest: "{{ rollback_sql_path }}"
+    remote_src: yes
```

Comment:
Why is this step needed? Can't we use this path as the dump's target?

Reply:
This was a workaround, as we have to use the `postgres` user to dump the db in the preceding step, and that user doesn't have the necessary folder permissions. I agree it seems a bit more convoluted, but it didn't work otherwise. 

I could take another go at it...

Reply:
Fair enough. Let's iterate

Reply:
I see. That also means that the backup will be visible to other system users. I don't think it's a big problem with our current setups but could be a security issue when sharing the server with other apps and users. Can we store it in the postgres user's home directory instead of tmp and then move it as root?

Reply:
I've changed this to dump the db to the default postgres user's home folder, and used `mv` as root immediately after, so the temporary file is removed.